### PR TITLE
Improved Hybrid renderer performance.

### PR DESCRIPTION
**Pull request by DBouma**

### CODE REVIEW

File: resources/shaders/dxr_global.hlsl [hunk 1]
```diff
@@ -24,7 +24,7 @@
 #define SOFT_SHADOWS
 #define MAX_SHADOW_SAMPLES 1
 
-//#define PERFECT_MIRROR_REFLECTIONS
+#define PERFECT_MIRROR_REFLECTIONS
```
Comment (Nielsbishere) [inline resources/shaders/dxr_global.hlsl]: No.

File: resources/shaders/dxr_reflection_functions.hlsl [hunk 2]
```diff
@@ -85,8 +85,9 @@ float4 DoReflection(float3 wpos, float3 V, float3 N, uint rand_seed, uint depth,
 
 		// Shoot perfect mirror ray if enabled or if it's a recursion or it's almost a perfect mirror
 
-		if (depth > 0 || roughness < 0.05f)
+		if (depth > 0 || roughness < 0.075f)
```
Comment (Nielsbishere) [inline resources/shaders/dxr_reflection_functions.hlsl]: Why decrease the roughness value required to get a perfect mirror?

File: resources/shaders/dxr_reflection_functions.hlsl [hunk 3]
```diff
@@ -165,9 +166,5 @@ float4 DoReflection(float3 wpos, float3 V, float3 N, uint rand_seed, uint depth,
 
 		#endif
 
-	#else
```
Comment (Nielsbishere) [inline resources/shaders/dxr_reflection_functions.hlsl]: Why remove this? It allows perfect mirrors.

**Review by Nielsbishere — CHANGES_REQUESTED** [review]

**Comment by Nielsbishere** [comment]
This doesn't even solve infinite recursions.

File: resources/shaders/dxr_global.hlsl (hunk 1 above)
Comment (161563) [inline resources/shaders/dxr_global.hlsl]: Doing this should disable rough reflections. PERFECT_MIRROR_REFLECTIONS shouldn't normally be defined.

File: resources/shaders/dxr_reflection_functions.hlsl [hunk 4]
```diff
@@ -85,8 +85,9 @@ float4 DoReflection(float3 wpos, float3 V, float3 N, uint rand_seed, uint depth,
 
 		// Shoot perfect mirror ray if enabled or if it's a recursion or it's almost a perfect mirror
 
-		if (depth > 0 || roughness < 0.05f)
+		if (depth > 0 || roughness < 0.075f)
 			return float4(TraceReflectionRay(wpos, N, reflected, rand_seed, depth, cone, dir_t), 1);
+	#endif
```
Comment (161563) [inline resources/shaders/dxr_reflection_functions.hlsl]: This means that turning on PERFECT_MIRROR_REFLECTIONS will instead give you rough reflections, including for recursions, which means the define now does the opposite of what it should do.

**Review by 161563 — CHANGES_REQUESTED** [review]
How does this fix stop infinite recursions from happening? None of these changes suggest that they fixed it, or even that we've had them in the first place.

File: resources/shaders/dxr_global.hlsl (hunk 1 above)
Comment (DBouma) [reply resources/shaders/dxr_global.hlsl]: it does not.

**Pull request closed without merging**